test/c/src/cross_serialization_tests.cpp

```diff
@@ -0,0 +1,175 @@
+#include "rosidl_typesupport_introspection_c/field_types.h"
+
+#include <rosidl_runtime_c/string.h>
+#include <rosidl_runtime_c/string_functions.h>
+
+typedef struct genericSequence {
+  void * data;
+  size_t size;
+  size_t capacity;
+} genericSequence;
+
+template <typename T>
+class CrossSerialization {
+  public:
+
+  #define QUICK_RETURN(x) if(!(x)) return x;
+
+  CrossSerialization(const rosidl_message_type_support_t * type_support_c) :
+    xrce_buffer(nullptr),
+    xrce_size(0),
+    fast_buffer(nullptr),
+    fast_size(0)
+  {
+    xrce_typesupport = get_message_typesupport_handle(type_support_c, rosidl_typesupport_microxrcedds_c__identifier);
+    fast_typesupport = get_message_typesupport_handle(type_support_c, rosidl_typesupport_fastrtps_c__identifier);
+    introspection_typesupport = get_message_typesupport_handle(type_support_c, rosidl_typesupport_introspection_c__identifier);
+
+    xrce_callbacks = static_cast<const message_type_support_callbacks_xrce_t *>(xrce_typesupport->data);
+    fast_callbacks = static_cast<const message_type_support_callbacks_fast_t *>(fast_typesupport->data);
+    introspection = (rosidl_typesupport_introspection_c__MessageMembers*) introspection_typesupport->data;
+  };
+
+  bool check() {
+    bool ret = true;
+    ret &= xrce_typesupport != nullptr;
+    ret &= fast_typesupport != nullptr;
+    ret &= introspection_typesupport != nullptr;
+    ret &= xrce_callbacks != nullptr;
+    ret &= fast_callbacks != nullptr;
+    ret &= introspection != nullptr;
+    return ret;
+  }
+
+  ~CrossSerialization() {
+    if(nullptr != xrce_buffer) {
+      free(xrce_buffer);
+    }
+    if(nullptr != fast_buffer) {
+      free(fast_buffer);
+    }
+  }
+
+  bool serialize_and_compare_buffers(T & msg) {
+    xrce_size = xrce_callbacks->get_serialized_size(&msg);
+    fast_size = fast_callbacks->get_serialized_size(&msg);
+    QUICK_RETURN(xrce_size == fast_size);
+
+    // size_t xrce_max_size = xrce_callbacks->max_serialized_size();
+    // bool full_bounded;
+    // size_t fast_max_size = fast_callbacks->max_serialized_size(full_bounded);
+    // QUICK_RETURN(xrce_max_size == fast_max_size);
+
+    xrce_buffer = (uint8_t *) calloc(xrce_size, sizeof(uint8_t));
+    ucdrBuffer xrce_cdr;
+    ucdr_init_buffer(&xrce_cdr, xrce_buffer, xrce_size);
+    QUICK_RETURN(xrce_callbacks->cdr_serialize(&msg, &xrce_cdr));
+
+    fast_buffer = (uint8_t *) calloc(fast_size, sizeof(uint8_t));
+    eprosima::fastcdr::FastBuffer fast_cdr_buffer((char *) fast_buffer, fast_size);
+    eprosima::fastcdr::Cdr fast_cdr(fast_cdr_buffer);
+    QUICK_RETURN(fast_callbacks->cdr_serialize(&msg, fast_cdr));
+
+    QUICK_RETURN(xrce_cdr.offset == fast_cdr.getSerializedDataLength());
+
+    bool equal_buffers = true;
+    for(size_t i = 0; i < xrce_size && equal_buffers; ++i){
+      equal_buffers &= xrce_buffer[i] == fast_buffer[i];
+      if(!equal_buffers) {
+        std::cout << "Error at position " << i << std::endl;
+      }
+    }
```

Comment:
Can we use `memcmp` here?

Reply:
Addressed 0c1d8b5c29521a4f205c5532d4bcb9fcd87cb84f